src/hooks/use-aggrid-row-filter.ts

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import { useCallback, useMemo, useState } from 'react';
+
+type FilterDataType = { value: string; type: string; dataType: string };
+
+type FilterType = {
+    field: string;
+    data: FilterDataType;
+};
+
+export type FilterEnumsType = Record<string, string[] | null>;
+
+export type FilterSelectorType = {
+    column: string;
+    dataType: string;
+    type: string;
+    value: string;
+};
+
+export type FilterPropsType = {
+    updateFilter: (field: string, value: FilterDataType) => void;
+    filterSelector: FilterSelectorType[] | null;
+    initFilters?: () => void;
+};
+
+type UseAggridRowFilterOutputType = {
+    updateFilter: (field: string, data: FilterDataType) => void;
+    filterSelector: FilterSelectorType[] | null;
+    initFilters: () => void;
+};
+
+const removeElementFromArrayWithFieldValue = (
+    filtersArrayToRemoveFieldValueFrom: FilterType[],
+    field: string
+) => {
+    return filtersArrayToRemoveFieldValueFrom.filter(
+        (f: FilterType) => f.field !== field
+    );
+};
+
+const changeValueFromArrayWithFieldValue = (
+    filtersArrayToModify: FilterType[],
+    field: string,
+    newData: FilterDataType
+) => {
+    const filterIndex = filtersArrayToModify.findIndex(
+        (f: FilterType) => f.field === field
+    );
+    if (filterIndex === -1) {
```

Comment:
you can use SORT_WAYS.desc

Reply:
it is not related to sort.. It is juste the index of a member that not exist in a array
(if the member with filed === field do not exist in filtersArrayToModify => findIndex will return -1)

Reply:
My bad